Framework/Algorithms/src/GeneratePeaks.cpp

```diff
@@ -767,7 +767,7 @@ GeneratePeaks::createDataWorkspace(std::vector<double> binparameters) {
   MatrixWorkspace_sptr ws = API::WorkspaceFactory::Instance().create(
       "Workspace2D", m_spectraSet.size(), numxvalue, numxvalue - 1);
   for (size_t ip = 0; ip < m_spectraSet.size(); ip++)
-    std::copy(xarray.begin(), xarray.end(), ws->dataX(ip).begin());
+    std::copy(xarray.begin(), xarray.end(), ws->mutableX(ip).begin());
```

Comment:
For creating `edges`: Actually last night Andrei merged a PR of mine that adds `HistogramData/LinearGenerator` and `HistogramData/LogarithmicGenerator`. Those should make it possible to replace the code block above that creates `xarray`. You should be able to find plenty of code examples where I am using those generators, but if you have trouble let me know!